cli/bptest/metadata_lint.go

```diff
@@ -0,0 +1,114 @@
+package bptest
+
+import (
+	"fmt"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/cli/bpmetadata"
+	"google.golang.org/protobuf/encoding/protojson"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/yaml"
+)
+
+// RunMetadataLintCommand is the entry function that will run the metadata.yml lint checks.
+func RunMetadataLintCommand() {
+	metadataFile := findMetadataFile()
+	if metadataFile == "" {
+		fmt.Println("Error: metadata.yaml file not found")
+		os.Exit(1)
+	}
+
+	metadata, err := ParseMetadataFile(metadataFile)
+	if err != nil {
+		fmt.Printf("Error parsing metadata file: %v\n", err)
+		os.Exit(1)
+	}
+
+	ctx := LintContext{
+		Metadata: metadata,
+		FilePath: metadataFile,
+	}
+
+	runner := &LintRunner{}
+	runner.RegisterRule(&BlueprintVersionRule{})
+
+	// Run lint checks
+	errs := runner.Run(ctx)
+	if len(errs) > 0 {
+		fmt.Println("Linting failed with the following errors:")
+		for _, err := range errs {
+			fmt.Println("- ", err)
+		}
+		os.Exit(1)
+	} else {
+		fmt.Println("All lint checks passed!")
+	}
+}
+
+// findMetadataFile searches for 'metadata.yaml' in the current directory or parent directories.
```

Comment:
I might be missing some context but why seach up parent dirs? This is commonly executed at rootmodule so IMO the general pattern is search down child dirs (and we should be careful not to avoid looking at .terraform dirs).

We also have https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/cli/util/file.go#L51 that you could use